### PR TITLE
fix(edit-profile): correct validation, add char counter, add corrections

### PR DESCRIPTION
**Pull request by anette1983**
## Describe your changes

- Corrected edit-profile form validation.
- Added character counter
- Fixed bugs

image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://ticket-url)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AKutusenko — APPROVED** [review]

File: src/components/hook-form/rhf-text-area.tsx [hunk 1]
```diff
@@ -1,41 +1,71 @@
...
+import { colors } from 'constants/colors';
 
 type Props = TextFieldProps & {
   name: string;
 };
 
 export default function RHFTextArea({ name, helperText, type, ...other }: Props): JSX.Element {
   const { control } = useFormContext();
+  const MAX_CHARS = 300;
+  const CHAR_PART = 0.05;
```
Comment (lyudad) [inline src/components/hook-form/rhf-text-area.tsx]: зачем переиниализировать константы при каждом ререндере?

Comment (anette1983) [reply src/components/hook-form/rhf-text-area.tsx]: My bad, sorry, fixed

File: src/components/hook-form/rhf-text-area.tsx [hunk 2]
```diff
@@ -8,34 +12,59 @@ type Props = TextFieldProps & {
 export default function RHFTextArea({ name, helperText, type, ...other }: Props): JSX.Element {
   const { control } = useFormContext();
 
+  const [charCount, setCharCount] = useState<number>(values.MAX_CHARS);
+  const color = charCount < values.CHAR_PART * values.MAX_CHARS ? colors.ERROR_COLOR : null;
```
Comment (lyudad) [inline src/components/hook-form/rhf-text-area.tsx]: если что то ставишь в константы то мемоизируй, а то это же при каждом ререндере будет пересчитыватся

Comment (anette1983) [reply src/components/hook-form/rhf-text-area.tsx]: Поняла, исправила

**Review by lyudad — APPROVED** [review]

**Pull request merged**